frontend/src/Commons/Button.tsx

```diff
@@ -1,7 +1,17 @@
 import StyledButton from './Styles/Button.style';
 
-const Button = (
-  { text }: { text: string }, // Destructured primitive string literal
-) => <StyledButton type="button">{text}</StyledButton>;
+export type ButtonSizes = 'sm' | 'md' | 'lg';
+export type ButtonProps = {
+  handleClick?: () => void;
+  text: string;
+  size: ButtonSizes;
+  disabled: boolean;
+};
+
+const Button = ({ handleClick, text, size = 'md', disabled = false }: ButtonProps) => (
```

Comment:
Shouldn't we add a default value for handleClick here? this might break stuff if we pass undefined to `onClick`. I'd add a ternary to check for nill and ignore it if that's the case. Something like:

```typescript
const Button = ({ handleClick=nill, text, size = 'md', disabled = false }: ButtonProps) => (
const defaultHandleClick = () => {
   console.log("default clickity click");
}
  <StyledButton size={size} disabled={disabled} onClick={handleClick ? handleClick : defaultHandleClick} value="">
    {text}
  </StyledButton>
);
```
I think thats what that eslint rule is for, although sometimes the linter still complains if we don't add an extra `type.defaultProps` on the bottom... I dealt with this and could give a hand if you need. But we should definitly check for nill here.